File: src/reviews/routes/reviewRoutes.js
const express = require('express');
const Review = require('../models/review');
const router = express.Router();

// Submit Review
router.post('/', async (req, res) => {
  try {
    const { product, user, rating, comment } = req.body;
    const newReview = await Review.create({ product, user, rating, comment });
    res.status(201).send(newReview);
  } catch (error) {
    res.status(400).send(error);
  }
});

// Update Review
router.put('/:id', async (req, res) => {
  try {
    const { rating, comment } = req.body;
    const updatedReview = await Review.findByIdAndUpdate(req.params.id, { rating, comment }, { new: true });
    res.status(200).send(updatedReview);
  } catch (error) {
    res.status(400).send(error);
  }
});

// Delete Review
router.delete('/:id', async (req, res) => {
  try {
    await Review.findByIdAndDelete(req.params.id);
    res.status(204).send();
  } catch (error) {
    res.status(500).send(error);
  }
});

// Get Product Reviews
router.get('/product/:productId', async (req, res) => {
  try {
    const reviews = await Review.find({ product: req.params.productId });
    res.status(200).send(reviews);
  } catch (error) {
    res.status(500).send(error);
  }
});

// Get Customer Reviews
router.get('/user/:userId', async (req, res) => {
  try {
    const reviews = await Review.find({ user: req.params.userId });
    res.status(200).send(reviews);
  } catch (error) {
    res.status(500).send(error);
  }
});

module.exports = router;

File: src/reviews/routes/reviewRoutes.py
from flask import Blueprint, request, jsonify
from db import db
from review import Review

review_bp = Blueprint('reviews', __name__, url_prefix='/reviews')

@review_bp.route('/', methods=['POST'])
def submit_review():
    data = request.get_json()
    new_review = Review(product_id=data['product'], user_id=data['user'],
                        rating=data['rating'], comment=data['comment'])
    db.session.add(new_review)
    db.session.commit()
    return jsonify(new_review), 201

@review_bp.route('/<int:review_id>', methods=['PUT'])
def update_review(review_id):
    data = request.get_json()
    review = Review.query.get(review_id)
    if review:
        review.rating = data.get('rating', review.rating)
        review.comment = data.get('comment', review.comment)
        db.session.commit()
        return jsonify(review), 200
    else:
        return jsonify({'error': 'Review not found'}), 404

@review_bp.route('/<int:review_id>', methods=['DELETE'])
def delete_review(review_id):
    review = Review.query.get(review_id)
    if review:
        db.session.delete(review)
        db.session.commit()
        return '', 204
    else:
        return jsonify({'error': 'Review not found'}), 404

@review_bp.route('/product/<int:product_id>', methods=['GET'])
def get_product_reviews(product_id):
    reviews = Review.query.filter_by(product_id=product_id).all()
    return jsonify(reviews), 200

@review_bp.route('/user/<int:user_id>', methods=['GET'])
def get_user_reviews(user_id):
    reviews = Review.query.filter_by(user_id=user_id).all()
    return jsonify(reviews), 200
